src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -450,16 +459,23 @@ def find_fixed_range(uworker_input):
             clear_min_max_metadata=clear_min_max_metadata),
         error_type=uworker_msg_pb2.ErrorType.PROGRESSION_BUILD_NOT_FOUND)
 
+  # The calls to `check_for_bad_build` return BuildData even when the build is
+  # not bad. We add all such BuildData information to this list and pass it to
+  # postprocess.
+  build_data_list = []
+
   # Check to see if this testcase is still crashing now. If it is, then just
   # bail out.
   result, error = _testcase_reproduces_in_revision(
       testcase,
       testcase_file_path,
       job_type,
       max_revision,
+      build_data_list,
       update_metadata=True,
       clear_min_max_metadata=clear_min_max_metadata)
   if error is not None:
+    error.progression_task_output.build_data_list.extend(build_data_list)
```

Comment:
Does assigning to a list instead of extending break things?

Reply:
yes, and I think the issue comes from here:
https://github.com/google/clusterfuzz/blob/4c9ca5e17c5d95f58bd761a1cf2204f8ceb8d3be/src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py#L438
Where `save_json_field` doesn't seem to be working for lists in the current state.

On the other hand, we can't assign lists of primitive types either but the error message is different.
Example for bad_revisions:
```
Traceback (most recent call last):
  File "/usr/local/google/home/ahijazi/alhijazi-clusterfuzz/clusterfuzz/src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/progression_task_test.py", line 177, in test_preprocess_uworker_output_custom_binary
    str(testcase.key.id()), 'job_type', None)
  File "/usr/local/google/home/ahijazi/alhijazi-clusterfuzz/clusterfuzz/src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py", line 384, in utask_preprocess
    progression_input.bad_revisions=build_manager.get_job_bad_revisions()
  File "/usr/local/google/home/ahijazi/alhijazi-clusterfuzz/clusterfuzz/src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py", line 352, in __setattr__
    setattr(self.proto, attribute, value)
AttributeError: Assignment not allowed to repeated field "bad_revisions" in protocol message object.
```

Reply:
I'll take a look at how we serialize lists of objects tomorrow and see if a fix is needed somewhere.
It seems for the moment that we properly handle serialization of lists in UworkerInput but only for `ndb.Model` instances.
https://github.com/google/clusterfuzz/blob/4c9ca5e17c5d95f58bd761a1cf2204f8ceb8d3be/src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py#L456